pcdsdevices/utils.py

```diff
@@ -687,3 +690,151 @@ def post_ophyds_to_elog(objs, allow_child=False, hutch_elog=None):
+    """
+    if cpts_or_names is None:
+        return []
+    reverse_map = {cpt: name for name, cpt in cls._sig_attrs.items()}
+    output = []
+    for obj in cpts_or_names:
+        if isinstance(obj, Cpt):
+            output.append(reverse_map[obj])
+        else:
+            output.append(obj)
+    return output
+
+
+def move_subdevices_to_start(cls: type, subdevice_cls: type = Device):
+    """
+    Arrange the component order of a device class to put subdevices first.
+
+    This can be useful to bring e.g. all the motors to the top for the
+    typhos screen.
+
+    The relative ordering of subdevices is preserved.
+
+    Parameters
+    ----------
+    cls : type
+        The Device subclass that we'd like to rearrange the order of.
+    subdevice_cls: type, optional
+        A specific class type to move to the front. If omitted, all device
+        subclasses will be moved.
+    """
+    device_names = []
+    for name, cpt in cls._sig_attrs.items():
+        if issubclass(cpt.cls, subdevice_cls):
+            device_names.append(name)
+    reorder_components(cls, start_with=device_names)
+
+
+def sort_components_by_name(cls: type, reverse: bool = False):
+    """
+    Arrange the component order of a device class in alphabetical order.
+
+    This can be useful as a first step before bringing specific components
+    to the top of the queue for the typhos screen.
+
+    Parameters
+    ----------
+    cls : type
+        The Device subclass that we'd like to rearrange the order of.
+    reverse : bool, optional
+        Set to True to sort in descending order instead.
+    """
+    alphabetical = list(sorted(cls._sig_attrs, reverse=reverse))
+    reorder_components(cls, start_with=alphabetical)
+
+
+def sort_components_by_kind(cls: type):
+    """
+    Arrange the component order of a device class in kind order.
+
+    Kind order is hinted > normal > config > omitted.
+
+    This can be useful because typically the higher kind classes
+    are more important, and therefore should be higher up on the
+    typhos screen.
+
+    The relative ordering of subdevices within a kind is preserved.
+
+    Parameters
+    ----------
+    cls : type
+        The Device subclass that we'd like to rearrange the order of.
+    """
+    hinted = []
+    normal = []
+    config = []
+    omitted = []
+    for name, cpt in cls._sig_attrs.items():
+        if cpt.kind is Kind.hinted:
+            hinted.append(name)
+        elif cpt.kind is Kind.normal:
+            normal.append(name)
+        elif cpt.kind is Kind.config:
+            config.append(name)
+        elif cpt.kind is Kind.omitted:
+            omitted.append(name)
+        else:
+            raise TypeError(f'Invalid component kind {cpt.kind}')
+    reorder_components(cls, end_with=hinted)
+    reorder_components(cls, end_with=normal)
+    reorder_components(cls, end_with=config)
+    reorder_components(cls, end_with=omitted)
+
+
+def set_standard_ordering(cls: type):
```

Comment:
`set_standard_ordering`, `sort_components_by_kind`, and others that don't take arguments outside of `cls` could easily be converted into class decorators. They would, of course, work as tested here as well.

May or may not be worth it, but it is an easy change.

Reply:
For this to work, all I have to do is make them return `cls`, right?

Can't we do this with all of them? I could have sworn there was a way to do something like:
```
@reorder_components(starts_with=['motor'])
class MyClass(Device):
    ...
```

Reply:
Yes, you can do that as well, just the decorator body gets a bit clunkier. No idea about appropriate annotations, but a common way I've seen/used looks like:

```python
def wrapper(cls=None, **kwargs):
    def wrap_the_class(cls):
        ...  # access kwargs here in the closure
        return cls
    
    if cls is not None:
        # wrapper(cls, starts_with=["motor"])
        return wrap_the_class(cls)
    # @wrapper(**kwargs) decorator on class cls
    return wrap_the_class
```